superduperdb/backends/base/query.py

```diff
@@ -145,6 +145,33 @@ def _set_db(r, db):
             parts.append((part, part_args, part_kwargs))
         self.parts = parts
 
+    def listeners(self, ):
+        listeners = self.db.show('listener')
+        vector_indices = self.db.show('vector_index')
+        dependencies = []
+        def _check_query_match(listener, query):
+            if (
+                listener.select.table_or_collection.identifier
+                == query.table_or_collection.identifier
+            ):
+                return True
+            return False
+
+        
+        for listener in listeners:
```

Comment:
Maybe now is the time to introduce triggers as a mixin or base class for `Listener` and `VectorIndex`?

Reply:
may be as separate pr